apps/fz_http/lib/fz_http/devices.ex

```diff
@@ -84,17 +95,22 @@ defmodule FzHttp.Devices do
 
   def delete_device(%Device{} = device) do
     Telemetry.delete_device()
-    Repo.delete(device)
+
+    case Repo.delete(device) do
+      {:ok, device} ->
+        Events.delete(device)
+        {:ok, device}
+
+      err ->
+        err
+    end
```

Comment:
`with` would be nice here because we don't match on errors:
```suggestion
    with {:ok, device} <- Repo.delete(device) do
      Events.delete(device)
      {:ok, device}
    end
```